xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp

```diff
@@ -177,102 +182,99 @@ void CGUIDialogVideoManagerVersions::SetDefaultVideoVersion(const CFileItem& ver
   CServiceBroker::GetGUI()->GetWindowManager().SendMessage(msg);
 }
 
-void CGUIDialogVideoManagerVersions::AddVideoVersion()
+bool CGUIDialogVideoManagerVersions::AddVideoVersion()
 {
-  const int dbId{m_videoAsset->GetVideoInfoTag()->m_iDbId};
-  const MediaType mediaType{m_videoAsset->GetVideoInfoTag()->m_type};
-  const VideoDbContentType itemType{m_videoAsset->GetVideoContentType()};
-
-  // prompt to choose a video file
-  VECSOURCES sources{*CMediaSourceSettings::GetInstance().GetSources("files")};
+  CVideoDatabase videoDb;
+  if (!videoDb.Open())
+  {
+    CLog::LogF(LOGERROR, "Failed to open video database!");
+    return false;
+  }
 
-  CServiceBroker::GetMediaManager().GetLocalDrives(sources);
-  CServiceBroker::GetMediaManager().GetNetworkLocations(sources);
+  CFileItemList items;
+  if (!GetSimilarMovies(m_videoAsset, items, videoDb))
+    return false;
 
-  std::string path;
-  if (CGUIDialogFileBrowser::ShowAndGetFile(
-          sources, CServiceBroker::GetFileExtensionProvider().GetVideoExtensions(),
-          g_localizeStrings.Get(40014), path))
+  if (items.Size() == 0)
   {
-    std::string typeVideoVersion;
-    std::string videoTitle;
-    int idFile{-1};
-    int idMedia{-1};
-    MediaType itemMediaType;
-    VideoAssetType videoAssetType{VideoAssetType::UNKNOWN};
+    // No button = browse library
+    // Yes button = browse files
+    // Custom button = cancel
 
-    const int idVideoVersion{m_database.GetVideoVersionInfo(path, idFile, typeVideoVersion, idMedia,
-                                                            itemMediaType, videoAssetType)};
+    const int dlgResult = CGUIDialogYesNo::ShowAndGetInput(
```

Comment:
initializer vs. assignment

Reply:
Sure. ChatGPT, please rewrite my PR in the style of @ksooo :-)